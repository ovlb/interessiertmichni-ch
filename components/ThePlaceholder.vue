<template>
  <div
    aria-label="Bitte warten"
    class="placeholder"
  >
    <span
      aria-hidden="true"
      class="shruggy"
    >
      <span class="shruggy--part shruggy--left">¯\_</span>
      <span class="shruggy--part shruggy--center">(ツ)</span>
      <span class="shruggy--part shruggy--right">_/¯</span>
    </span>
  </div>
</template>

<script>
export default {

}
</script>

<style>

@keyframes shruggyLeft {
  from {
    transform: translateY(0);
    transform-origin: left bottom;
  }

  50% {
    transform: translateY(-3%) rotate(-2deg);
    transform-origin: left bottom;
  }

  to {
    transform-origin: left bottom;
    transform: translateY(0);
  }
}

@keyframes shruggyRight {
  from {
    transform-origin: right bottom;
    transform: translateY(0);
  }

  50% {
    transform: translateY(-3.33333%) rotate(2deg);
    transform-origin: right bottom;
  }

  to {
    transform-origin: right bottom;
    transform: translateY(0);
  }
}

.shruggy {
  display: flex;
}

.shruggy--part {
  display: block;
}

.shruggy--left {
  animation: shruggyLeft 1s linear infinite both;
}

.shruggy--right {
  transform-origin: left bottom;
  animation: shruggyRight 1s linear infinite both;
}

.placeholder {
  align-items: center;
  background: aliceblue;
  height: 66vmin;
  display: flex;
  font-size: 3rem;
  justify-content: center;
  max-height: calc(.66 * 35rem);
}

</style>
