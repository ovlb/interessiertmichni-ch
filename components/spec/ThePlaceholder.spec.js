import { shallow } from '@vue/test-utils'
import ThePlaceholder from '../ThePlaceholder.vue'

describe('ThePlaceholder', () => {
  const wrapper = shallow(ThePlaceholder)

  test('match HTML sturcture', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
