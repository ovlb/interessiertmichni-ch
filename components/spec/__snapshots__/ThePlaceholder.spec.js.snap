// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThePlaceholder match HTML sturcture 1`] = `
<div
  aria-label="Bitte warten"
  class="placeholder"
>
  <span
    aria-hidden="true"
    class="shruggy"
  >
    <span
      class="shruggy--part shruggy--left"
    >
      ¯\\_
    </span>
     
    <span
      class="shruggy--part shruggy--center"
    >
      (ツ)
    </span>
     
    <span
      class="shruggy--part shruggy--right"
    >
      _/¯
    </span>
  </span>
</div>
`;
